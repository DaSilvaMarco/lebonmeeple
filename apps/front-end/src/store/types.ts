// Types centralisés pour le store Redux

import type { RootState } from './index';

// Sélecteurs typés avec RootState
export const createTypedSelector = <T>(selector: (state: RootState) => T) =>
  selector;

// Types d'export pour les autres fichiers
export type { RootState } from './index';
export type { AppDispatch } from './index';

export interface User {
  id: string;
  username: string;
  email: string;
  avatar?: string;
  createdAt?: string;
}

export interface AuthState {
  user: User | null;
  token: string | null;
  isAuthenticated: boolean;
  isLoading: boolean;
}

export const initialState: AuthState = {
  user: null,
  token: null,
  isAuthenticated: false,
  isLoading: false,
};
