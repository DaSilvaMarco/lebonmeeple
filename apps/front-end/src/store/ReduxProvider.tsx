'use client';

import React, { useEffect } from 'react';
import { Provider, useDispatch } from 'react-redux';
import { store } from '@/store/store';
import { login } from '@/domains/user/slice';

type ReduxProviderProps = {
  children: React.ReactNode;
};

function UserHydrator({ children }: { children: React.ReactNode }) {
  const dispatch = useDispatch();
  useEffect(() => {
    const token =
      typeof window !== 'undefined' ? localStorage.getItem('authToken') : null;
    const userStr =
      typeof window !== 'undefined' ? localStorage.getItem('user') : null;
    if (token && userStr) {
      try {
        const user = JSON.parse(userStr);
        dispatch(login({ user, token }));
      } catch (error) {
        console.log(error);
      }
    }
  }, [dispatch]);
  return <>{children}</>;
}

export function ReduxProvider({ children }: ReduxProviderProps) {
  return (
    <Provider store={store}>
      <UserHydrator>{children}</UserHydrator>
    </Provider>
  );
}
