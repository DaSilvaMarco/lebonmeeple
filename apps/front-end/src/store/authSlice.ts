import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { type AuthState, initialState, type User } from './types';

const authSlice = createSlice({
  name: 'auth',
  initialState,
  reducers: {
    login: (state, action: PayloadAction<{ user: User; token: string }>) => {
      state.isLoading = false;
      state.user = action.payload.user;
      state.token = action.payload.token;
      state.isAuthenticated = true;

      localStorage.setItem('authToken', action.payload.token);
      localStorage.setItem('user', JSON.stringify(action.payload.user));
    },

    logout: (state) => {
      state.user = null;
      state.token = null;
      state.isAuthenticated = false;
      state.isLoading = false;

      localStorage.removeItem('authToken');
      localStorage.removeItem('user');
    },

    updateUser: (state, action) => {
      console.log('updateUser payload:', action.payload);
      state.user = action.payload;
    },
  },
});

export const { login, logout, updateUser } = authSlice.actions;

export const selectUser = (state: { auth: AuthState }) => state.auth.user;
export const selectToken = (state: { auth: AuthState }) => state.auth.token;
export const selectIsAuthenticated = (state: { auth: AuthState }) =>
  state.auth.isAuthenticated;
export const selectIsLoading = (state: { auth: AuthState }) =>
  state.auth.isLoading;

export default authSlice.reducer;
