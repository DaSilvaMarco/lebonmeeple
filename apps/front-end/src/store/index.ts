import { configureStore } from '@reduxjs/toolkit';
import authReducer from './authSlice';

// Configuration du store Redux
export const store = configureStore({
  reducer: {
    auth: authReducer,
  },
  // Middleware par défaut avec serialization check désactivé pour les fonctions
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        // Ignorer ces actions qui peuvent contenir des fonctions
        ignoredActions: ['persist/PERSIST', 'persist/REHYDRATE'],
      },
    }),
});

// Types pour TypeScript
export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

// Hooks typés pour TypeScript
export { useDispatch, useSelector } from 'react-redux';
export type { TypedUseSelectorHook } from 'react-redux';

// Hooks personnalisés avec types
import {
  useDispatch as useReduxDispatch,
  useSelector as useReduxSelector,
  TypedUseSelectorHook,
} from 'react-redux';

export const useAppDispatch = () => useReduxDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useReduxSelector;
