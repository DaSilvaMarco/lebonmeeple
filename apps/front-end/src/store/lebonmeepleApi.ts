import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
import { getApiBaseUrl } from '@/utils/api-config';

export const lebonmeepleApi = createApi({
  reducerPath: 'lebonmeepleApi',
  baseQuery: fetchBaseQuery({ baseUrl: getApiBaseUrl() }),
  endpoints: () => ({}),
  tagTypes: ['Posts'],
  keepUnusedDataFor: 1,
  refetchOnMountOrArgChange: true,
});
