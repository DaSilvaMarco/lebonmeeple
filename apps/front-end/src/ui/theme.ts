import { type ThemeConfig, extendTheme } from '@chakra-ui/react';

// Configuration du thème (mode sombre/clair)
const config: ThemeConfig = {
  initialColorMode: 'light',
  useSystemColorMode: false,
};

// Couleurs personnalisées
const colors = {
  brand: {
    50: '#e3f2fd',
    100: '#bbdefb',
    200: '#90caf9',
    300: '#64b5f6',
    400: '#42a5f5',
    500: '#2196f3', // Couleur principale
    600: '#1e88e5',
    700: '#1976d2',
    800: '#1565c0',
    900: '#0d47a1',
  },
  // Couleurs pour Le Bon Meeple (thème jeu de société)
  meeple: {
    50: '#fff3e0',
    100: '#ffe0b3',
    200: '#ffcc80',
    300: '#ffb74d',
    400: '#ffa726',
    500: '#ff9800', // Orange principal
    600: '#fb8c00',
    700: '#f57c00',
    800: '#ef6c00',
    900: '#e65100',
  },
  game: {
    50: '#f3e5f5',
    100: '#e1bee7',
    200: '#ce93d8',
    300: '#ba68c8',
    400: '#ab47bc',
    500: '#9c27b0', // Violet pour le gaming
    600: '#8e24aa',
    700: '#7b1fa2',
    800: '#6a1b9a',
    900: '#4a148c',
  },
};

// Polices personnalisées
const fonts = {
  heading:
    '"Poppins", -apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, Arial, sans-serif',
  body: '"Inter", -apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, Arial, sans-serif',
};

// Styles des composants
const components = {
  // Button: {
  //   baseStyle: {
  //     fontWeight: 'semibold',
  //     borderRadius: 'lg',
  //   },
  //   variants: {
  //     solid: {
  //       bg: 'brand.500',
  //       color: 'white',
  //       _hover: {
  //         bg: 'brand.600',
  //         transform: 'translateY(-2px)',
  //         boxShadow: 'lg',
  //       },
  //       _active: {
  //         bg: 'brand.700',
  //         transform: 'translateY(0)',
  //       },
  //     },
  //     meeple: {
  //       bg: 'meeple.500',
  //       color: 'white',
  //       _hover: {
  //         bg: 'meeple.600',
  //         transform: 'translateY(-2px)',
  //         boxShadow: 'lg',
  //       },
  //       _active: {
  //         bg: 'meeple.700',
  //         transform: 'translateY(0)',
  //       },
  //     },
  //     game: {
  //       bg: 'game.500',
  //       color: 'white',
  //       _hover: {
  //         bg: 'game.600',
  //         transform: 'translateY(-2px)',
  //         boxShadow: 'lg',
  //       },
  //       _active: {
  //         bg: 'game.700',
  //         transform: 'translateY(0)',
  //       },
  //     },
  //   },
  //   defaultProps: {
  //     variant: 'solid',
  //   },
  // },
  Card: {
    baseStyle: {
      container: {
        borderRadius: 'xl',
        boxShadow: 'sm',
        _hover: {
          boxShadow: 'md',
          transform: 'translateY(-2px)',
        },
        transition: 'all 0.2s',
      },
    },
  },
  Input: {
    variants: {
      filled: {
        field: {
          bg: 'gray.50',
          borderRadius: 'lg',
          _hover: {
            bg: 'gray.100',
          },
          _focus: {
            bg: 'white',
            borderColor: 'brand.500',
            boxShadow: '0 0 0 1px var(--chakra-colors-brand-500)',
          },
        },
      },
    },
    defaultProps: {
      variant: 'filled',
    },
  },
  Heading: {
    baseStyle: {
      fontWeight: 'bold',
    },
    variants: {
      page: {
        fontSize: { base: '2xl', md: '3xl' },
        color: 'gray.800',
        marginBottom: 6,
      },
      section: {
        fontSize: { base: 'xl', md: '2xl' },
        color: 'gray.700',
        marginBottom: 4,
      },
    },
  },
};

// Styles globaux
const styles = {
  global: {
    body: {
      bg: 'gray.50',
      color: 'gray.800',
    },
    '*::placeholder': {
      color: 'gray.400',
    },
    '*, *::before, &::after': {
      borderColor: 'gray.200',
    },
  },
};

// Breakpoints personnalisés
const breakpoints = {
  base: '0em',
  sm: '30em',
  md: '48em',
  lg: '62em',
  xl: '80em',
  '2xl': '96em',
};

// Création du thème final
const theme = extendTheme({
  config,
  colors,
  fonts,
  components,
  styles,
  breakpoints,
  space: {
    px: '1px',
    0.5: '0.125rem',
    1: '0.25rem',
    1.5: '0.375rem',
    2: '0.5rem',
    2.5: '0.625rem',
    3: '0.75rem',
    3.5: '0.875rem',
    4: '1rem',
    5: '1.25rem',
    6: '1.5rem',
    7: '1.75rem',
    8: '2rem',
    9: '2.25rem',
    10: '2.5rem',
    12: '3rem',
    14: '3.5rem',
    16: '4rem',
    20: '5rem',
    24: '6rem',
    28: '7rem',
    32: '8rem',
    36: '9rem',
    40: '10rem',
    44: '11rem',
    48: '12rem',
    52: '13rem',
    56: '14rem',
    60: '15rem',
    64: '16rem',
    72: '18rem',
    80: '20rem',
    96: '24rem',
  },
});

export default theme;
