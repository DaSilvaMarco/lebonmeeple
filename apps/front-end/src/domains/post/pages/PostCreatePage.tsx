'use client';

import React from 'react';
import { Flex, Box } from '@chakra-ui/react';

import { PostCreateFormCard } from '../components';
import NotConnected from '@frontend/domains/shared/warning/NotConnected';
import { useAppSelector } from '@frontend/store/hook';

const PostCreatePage = () => {

  const { user, isAuthenticated } = useAppSelector((state) => state.user);

  if (!user || !isAuthenticated) {
    return <NotConnected />;
  }

  return (
    <Flex justify="center">
      <Box w="full" maxW={{ base: 'md', md: '2xl' }} zIndex={1}>
        <Box>
          <PostCreateFormCard />
        </Box>
      </Box>
    </Flex>
  );
};

export default PostCreatePage;
