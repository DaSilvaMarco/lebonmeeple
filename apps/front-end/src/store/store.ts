import { configureStore } from '@reduxjs/toolkit';
import userSlice from '@/domains/user/slice';
import { lebonmeepleApi } from './lebonmeepleApi';

export const store = configureStore({
  reducer: {
    user: userSlice,
    [lebonmeepleApi.reducerPath]: lebonmeepleApi.reducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(lebonmeepleApi.middleware),
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
