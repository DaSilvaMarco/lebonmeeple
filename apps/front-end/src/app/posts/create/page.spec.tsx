import React from 'react';
import { render, screen } from '@testing-library/react';
import { describe, it, expect, vi, beforeEach } from 'vitest';
import { Provider } from 'react-redux';
import { ChakraProvider } from '@chakra-ui/react';
import { configureStore } from '@reduxjs/toolkit';
import PostCreatePageApp from './page';
import { lebonmeepleApi } from '@frontend/store/lebonmeepleApi';
import theme from '@frontend/ui/theme';

// Mock du domaine post
vi.mock('@frontend/domains/post/pages/PostCreatePage', () => ({
  default: vi.fn(() => (
    <div data-testid="post-create-page">Mocked PostCreatePage</div>
  )),
}));

// Import du composant mocké pour les types
import PostCreatePage from '@frontend/domains/post/pages/PostCreatePage';

// Configuration du store de test
const createTestStore = () => {
  return configureStore({
    reducer: {
      [lebonmeepleApi.reducerPath]: lebonmeepleApi.reducer,
    },
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware().concat(lebonmeepleApi.middleware),
  });
};

// Composant wrapper pour les tests
const TestWrapper = ({ children }: { children: React.ReactNode }) => {
  const store = createTestStore();

  return (
    <Provider store={store}>
      <ChakraProvider theme={theme}>{children}</ChakraProvider>
    </Provider>
  );
};

describe('Post Create Page App - Complete Coverage', () => {
  beforeEach(() => {
    // Reset tous les mocks avant chaque test
    vi.clearAllMocks();

    // Remettre le mock par défaut pour PostCreatePage
    const MockedPostCreatePage = vi.mocked(PostCreatePage);
    MockedPostCreatePage.mockImplementation(() => (
      <div data-testid="post-create-page">Mocked PostCreatePage</div>
    ));
  });

  describe('Basic Functionality', () => {
    it('should render the page successfully', () => {
      render(
        <TestWrapper>
          <PostCreatePageApp />
        </TestWrapper>,
      );

      expect(screen.getByTestId('post-create-page')).toBeInTheDocument();
    });

    it('should render without any props', () => {
      const { container } = render(
        <TestWrapper>
          <PostCreatePageApp />
        </TestWrapper>,
      );

      expect(container).toBeDefined();
      expect(container.firstChild).toBeTruthy();
    });

    it('should call PostCreatePage component', () => {
      const MockedPostCreatePage = vi.mocked(PostCreatePage);

      render(
        <TestWrapper>
          <PostCreatePageApp />
        </TestWrapper>,
      );

      expect(MockedPostCreatePage).toHaveBeenCalledWith({}, {});
    });

    it('should call PostCreatePage exactly once', () => {
      const MockedPostCreatePage = vi.mocked(PostCreatePage);

      render(
        <TestWrapper>
          <PostCreatePageApp />
        </TestWrapper>,
      );

      expect(MockedPostCreatePage).toHaveBeenCalledTimes(1);
    });
  });

  describe('Component Structure', () => {
    it('should return JSX element', () => {
      const result = PostCreatePageApp();

      expect(result).toBeTruthy();
      expect(React.isValidElement(result)).toBe(true);
    });

    it('should be a functional component', () => {
      expect(typeof PostCreatePageApp).toBe('function');
    });

    it('should be synchronous (not async)', () => {
      expect(PostCreatePageApp.constructor.name).toBe('Function');
      expect(PostCreatePageApp.constructor.name).not.toBe('AsyncFunction');
    });

    it('should not require any parameters', () => {
      // La fonction ne devrait pas lever d'erreur sans paramètres
      expect(() => PostCreatePageApp()).not.toThrow();
    });

    it('should accept no props', () => {
      const result = PostCreatePageApp();
      expect(React.isValidElement(result)).toBe(true);
    });
  });

  describe('Domain Integration', () => {
    it('should integrate correctly with post domain', () => {
      render(
        <TestWrapper>
          <PostCreatePageApp />
        </TestWrapper>,
      );

      expect(screen.getByTestId('post-create-page')).toBeInTheDocument();
    });

    it('should maintain clean architecture', () => {
      const MockedPostCreatePage = vi.mocked(PostCreatePage);

      render(
        <TestWrapper>
          <PostCreatePageApp />
        </TestWrapper>,
      );

      // Vérifier que PostCreatePage est appelé correctement
      expect(MockedPostCreatePage).toHaveBeenCalledWith({}, {});
      expect(MockedPostCreatePage).toHaveBeenCalledTimes(1);
    });

    it('should delegate all logic to domain layer', () => {
      const MockedPostCreatePage = vi.mocked(PostCreatePage);

      render(
        <TestWrapper>
          <PostCreatePageApp />
        </TestWrapper>,
      );

      // Le composant application ne fait que déléguer
      expect(MockedPostCreatePage).toHaveBeenCalled();
    });
  });

  describe('Component Lifecycle', () => {
    it('should mount and unmount correctly', () => {
      const { unmount } = render(
        <TestWrapper>
          <PostCreatePageApp />
        </TestWrapper>,
      );

      expect(screen.getByTestId('post-create-page')).toBeInTheDocument();

      unmount();

      expect(screen.queryByTestId('post-create-page')).not.toBeInTheDocument();
    });

    it('should re-render without issues', () => {
      const { rerender } = render(
        <TestWrapper>
          <PostCreatePageApp />
        </TestWrapper>,
      );

      expect(screen.getByTestId('post-create-page')).toBeInTheDocument();

      rerender(
        <TestWrapper>
          <PostCreatePageApp />
        </TestWrapper>,
      );

      expect(screen.getByTestId('post-create-page')).toBeInTheDocument();
    });

    it('should handle multiple mounts', () => {
      const { unmount: unmount1 } = render(
        <TestWrapper>
          <PostCreatePageApp />
        </TestWrapper>,
      );

      expect(screen.getByTestId('post-create-page')).toBeInTheDocument();
      unmount1();

      const { unmount: unmount2 } = render(
        <TestWrapper>
          <PostCreatePageApp />
        </TestWrapper>,
      );

      expect(screen.getByTestId('post-create-page')).toBeInTheDocument();
      unmount2();
    });
  });

  describe('Code Quality', () => {
    it('should be a simple wrapper component', () => {
      const componentCode = PostCreatePageApp.toString();

      // Le composant ne devrait pas contenir de logique complexe
      expect(componentCode).not.toMatch(/if\s*\(/);
      expect(componentCode).not.toMatch(/switch/);
      expect(componentCode).not.toMatch(/for\s*\(/);
      expect(componentCode).not.toMatch(/while\s*\(/);
    });

    it('should follow single responsibility principle', () => {
      render(
        <TestWrapper>
          <PostCreatePageApp />
        </TestWrapper>,
      );

      // Le composant ne fait qu'une chose : rendre PostCreatePage
      expect(screen.getByTestId('post-create-page')).toBeInTheDocument();
    });

    it('should have minimal complexity', () => {
      const componentCode = PostCreatePageApp.toString();

      // Vérifier que c'est un composant simple
      expect(componentCode.split('\n').length).toBeLessThan(10);
    });

    it('should not contain business logic', () => {
      const componentCode = PostCreatePageApp.toString();

      // Pas de logique métier dans le composant application
      expect(componentCode).not.toMatch(/useState/);
      expect(componentCode).not.toMatch(/useEffect/);
      expect(componentCode).not.toMatch(/api/i);
      expect(componentCode).not.toMatch(/fetch/i);
      expect(componentCode).not.toMatch(/axios/i);
    });
  });

  describe('DDD Architecture Compliance', () => {
    it('should act as application layer component', () => {
      const MockedPostCreatePage = vi.mocked(PostCreatePage);

      render(
        <TestWrapper>
          <PostCreatePageApp />
        </TestWrapper>,
      );

      // Le composant application délègue au domaine
      expect(MockedPostCreatePage).toHaveBeenCalled();
    });

    it('should delegate business logic to domain layer', () => {
      const componentCode = PostCreatePageApp.toString();

      // Le composant ne devrait pas contenir de logique métier complexe
      expect(componentCode).not.toMatch(/business.*logic/i);
      expect(componentCode).not.toMatch(/validation/i);
      expect(componentCode).not.toMatch(/submit/i);
      expect(componentCode).not.toMatch(/create.*post/i);
    });

    it('should maintain separation of concerns', () => {
      const MockedPostCreatePage = vi.mocked(PostCreatePage);

      render(
        <TestWrapper>
          <PostCreatePageApp />
        </TestWrapper>,
      );

      // Vérification de la séparation des responsabilités
      expect(MockedPostCreatePage).toHaveBeenCalledWith({}, {});
    });

    it('should follow clean architecture principles', () => {
      const MockedPostCreatePage = vi.mocked(PostCreatePage);

      render(
        <TestWrapper>
          <PostCreatePageApp />
        </TestWrapper>,
      );

      // Le composant devrait déléguer au domaine PostCreatePage
      expect(MockedPostCreatePage).toHaveBeenCalled();

      // Vérifier qu'il n'y a pas de dépendances externes inappropriées
      const componentCode = PostCreatePageApp.toString();
      expect(componentCode).not.toMatch(/http/i);
      expect(componentCode).not.toMatch(/database/i);
      expect(componentCode).not.toMatch(/repository/i);
    });
  });

  describe('Props Interface and Types', () => {
    it('should work without any props', () => {
      const result = PostCreatePageApp();
      expect(React.isValidElement(result)).toBe(true);
    });

    it('should work with TypeScript strict mode', () => {
      // Test de compatibilité TypeScript
      const result: React.ReactElement = PostCreatePageApp();
      expect(React.isValidElement(result)).toBe(true);
    });

    it('should have correct function signature', () => {
      expect(PostCreatePageApp.length).toBe(0); // Aucun paramètre requis
    });

    it('should be compatible with React component interface', () => {
      // Vérifier que le composant peut être utilisé comme React component
      const TestComponent = () => <PostCreatePageApp />;

      render(
        <TestWrapper>
          <TestComponent />
        </TestWrapper>,
      );

      expect(screen.getByTestId('post-create-page')).toBeInTheDocument();
    });
  });

  describe('Component Naming and Export', () => {
    it('should export the component as default', () => {
      expect(PostCreatePageApp).toBeDefined();
      expect(typeof PostCreatePageApp).toBe('function');
    });

    it('should have correct component name', () => {
      expect(PostCreatePageApp.name).toBe('App');
    });

    it('should be importable as default export', () => {
      // Le composant devrait être exporté par défaut
      expect(PostCreatePageApp).toBeTruthy();
    });
  });

  describe('Next.js Page Integration', () => {
    it('should work as Next.js page component', () => {
      // Les pages Next.js retournent du JSX
      const result = PostCreatePageApp();
      expect(React.isValidElement(result)).toBe(true);
    });

    it('should be compatible with Next.js routing', () => {
      // Simuler l'utilisation par Next.js
      const result = PostCreatePageApp();
      expect(React.isValidElement(result)).toBe(true);
    });

    it('should handle Next.js page lifecycle', () => {
      const MockedPostCreatePage = vi.mocked(PostCreatePage);

      render(
        <TestWrapper>
          <PostCreatePageApp />
        </TestWrapper>,
      );

      expect(MockedPostCreatePage).toHaveBeenCalled();
    });

    it('should work with Next.js static routes', () => {
      // Test pour route statique /posts/create
      render(
        <TestWrapper>
          <PostCreatePageApp />
        </TestWrapper>,
      );

      expect(screen.getByTestId('post-create-page')).toBeInTheDocument();
    });
  });

  describe('Performance', () => {
    it('should render efficiently', () => {
      const startTime = performance.now();

      render(
        <TestWrapper>
          <PostCreatePageApp />
        </TestWrapper>,
      );

      const endTime = performance.now();
      const renderTime = endTime - startTime;

      // Vérifier que le rendu prend moins de 500ms
      expect(renderTime).toBeLessThan(500);
    });

    it('should handle multiple concurrent renders', () => {
      // Rendu concurrent
      const results = [
        PostCreatePageApp(),
        PostCreatePageApp(),
        PostCreatePageApp(),
      ];

      results.forEach((result) => {
        expect(React.isValidElement(result)).toBe(true);
      });
    });

    it('should not cause memory leaks', () => {
      const { unmount } = render(
        <TestWrapper>
          <PostCreatePageApp />
        </TestWrapper>,
      );

      expect(screen.getByTestId('post-create-page')).toBeInTheDocument();

      // Unmount ne devrait pas causer d'erreurs
      expect(() => unmount()).not.toThrow();
    });

    it('should be lightweight', () => {
      const componentString = PostCreatePageApp.toString();

      // Le composant devrait être très léger
      expect(componentString.length).toBeLessThan(1000);
    });
  });

  describe('Error Handling and Edge Cases', () => {
    it('should handle React context correctly', () => {
      render(
        <TestWrapper>
          <PostCreatePageApp />
        </TestWrapper>,
      );

      expect(screen.getByTestId('post-create-page')).toBeInTheDocument();
    });

    it('should work with different providers', () => {
      const CustomWrapper = ({ children }: { children: React.ReactNode }) => {
        const store = createTestStore();
        return (
          <Provider store={store}>
            <ChakraProvider theme={theme}>{children}</ChakraProvider>
          </Provider>
        );
      };

      render(
        <CustomWrapper>
          <PostCreatePageApp />
        </CustomWrapper>,
      );

      expect(screen.getByTestId('post-create-page')).toBeInTheDocument();
    });

    it('should handle provider changes gracefully', () => {
      const { rerender } = render(
        <TestWrapper>
          <PostCreatePageApp />
        </TestWrapper>,
      );

      expect(screen.getByTestId('post-create-page')).toBeInTheDocument();

      // Re-render avec un nouveau wrapper
      rerender(
        <TestWrapper>
          <PostCreatePageApp />
        </TestWrapper>,
      );

      expect(screen.getByTestId('post-create-page')).toBeInTheDocument();
    });

    it('should maintain component stability', () => {
      const MockedPostCreatePage = vi.mocked(PostCreatePage);

      // Premier rendu
      const { rerender } = render(
        <TestWrapper>
          <PostCreatePageApp />
        </TestWrapper>,
      );

      const firstCallCount = MockedPostCreatePage.mock.calls.length;

      // Re-render
      rerender(
        <TestWrapper>
          <PostCreatePageApp />
        </TestWrapper>,
      );

      // Le composant devrait être re-rendu
      expect(MockedPostCreatePage.mock.calls.length).toBeGreaterThan(
        firstCallCount,
      );
    });
  });

  describe('Integration with Testing Environment', () => {
    it('should work with React Testing Library', () => {
      render(
        <TestWrapper>
          <PostCreatePageApp />
        </TestWrapper>,
      );

      expect(screen.getByTestId('post-create-page')).toBeInTheDocument();
    });

    it('should work with Vitest mocking', () => {
      const MockedPostCreatePage = vi.mocked(PostCreatePage);

      render(
        <TestWrapper>
          <PostCreatePageApp />
        </TestWrapper>,
      );

      expect(MockedPostCreatePage).toHaveBeenCalled();
    });

    it('should be compatible with test environment', () => {
      // Test que le composant fonctionne dans l'environnement de test
      expect(() => {
        render(
          <TestWrapper>
            <PostCreatePageApp />
          </TestWrapper>,
        );
      }).not.toThrow();
    });

    it('should support snapshot testing', () => {
      const { container } = render(
        <TestWrapper>
          <PostCreatePageApp />
        </TestWrapper>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
