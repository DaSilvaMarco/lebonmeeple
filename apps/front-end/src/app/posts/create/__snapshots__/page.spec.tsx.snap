// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Post Create Page App - Complete Coverage > Integration with Testing Environment > should support snapshot testing 1`] = `
<div
  data-testid="post-create-page"
>
  Mocked PostCreatePage
</div>
`;
